wp-includes/components/class-wp-service-worker-navigation-routing-component.php

```diff
@@ -97,17 +97,44 @@ public function serve( WP_Service_Worker_Scripts $scripts ) {
 			 *
 			 * @param string $caching_strategy Caching strategy to use for frontend navigation requests.
 			 */
-			$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_ONLY );
+			$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_FIRST );
+
+			$caching_strategy_args = array(
+				'cacheName' => WP_Service_Worker_Caching_Routes::NAVIGATIONS_CACHE_NAME,
+			);
+			if ( WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_FIRST === $caching_strategy ) {
+				/*
+				 * The value of 2 seconds is informed by the Largest Contentful Paint (LCP) metric, of which Time to
+				 * First Byte (TTFB) is a major component. As long as all assets on a page are cached, then this allows
+				 * for the service worker to serve a previously-cached page and then for LCP to occur before 2.5s and
+				 * so remain within the good threshold.
+				 */
+				$caching_strategy_args['networkTimeoutSeconds'] = 2;
+			}
+
+			/*
+			 * By default cache only the last 10 pages visited. This may end up being too high as it seems likely that
+			 * most site visitors will view one page and then maybe a couple others.
+			 */
+			$caching_strategy_args['plugins']['expiration']['maxEntries'] = 10;
 
 			/**
 			 * Filters the caching strategy args used for frontend navigation requests.
 			 *
 			 * @since 0.2
+			 * @since 0.6 Added $caching_strategy param and initial array has default values provided.
 			 * @see WP_Service_Worker_Caching_Routes::register()
 			 *
-			 * @param array $caching_strategy_args Caching strategy args.
+			 * @param array $caching_strategy_args {
+			 *     Caching strategy args.
+			 *
+			 *     @type string $cacheName             Cache name to store navigation responses.
+			 *     @type int    $networkTimeoutSeconds Network timeout seconds when NetworkFirst strategy is used.
+			 *     @type array  $plugins               Configuration for plugins, in particular expiration.
```

Comment:
While Workbox has this `plugins` level of nesting, I think it would make sense to promote the children up a level as it would be less confusing when extending in a WordPress context. So instead of there being `plugins.expiration` there could be just `expiration`. Naturally we'd need to move into a `plugins` array when passing over to Workbox, but we already have a set of such known plugins when we `prepare_strategy_args_for_js_export`:

https://github.com/GoogleChromeLabs/pwa-wp/blob/ec984523a90d94afe49ebf65f0ec419b026322d4/wp-includes/components/class-wp-service-worker-caching-routes.php#L173-L179

So that would be easy to do.

Reply:
Done in efb519d3f47aa405e90c87d3842ddeae4f576b08.